components/ProgressView/src/MDCProgressView.m

```diff
@@ -109,12 +127,105 @@ - (void)setProgressTintColor:(UIColor *)progressTintColor {
     progressTintColor = MDCProgressViewDefaultTintColor();
   }
   self.progressView.backgroundColor = progressTintColor;
+  self.transitionView.backgroundColor = progressTintColor;
 }
 
 - (UIColor *)trackTintColor {
   return self.trackView.backgroundColor;
 }
 
+- (void)setMode:(MDCProgressViewMode)mode {
+  if (_mode == mode) {
+    return;
+  }
+  _mode = mode;
+
+  // If the progress bar was animating in indeterminate mode before, restart the animation.
+  if (_isAnimating && _mode == MDCProgressViewModeIndeterminate) {
+    [self.progressView.layer removeAllAnimations];
+    _isAnimating = false;
+    [self startAnimating];
+  }
+}
+
+- (void)setMode:(MDCProgressViewMode)mode
+       animated:(BOOL)animated
+     completion:(void (^__nullable)(BOOL finished))completion {
+  if (_mode == mode) {
+    return;
+  }
+  _mode = mode;
+
+  // Change from indeterminate to determinate.
+  if (_mode == MDCProgressViewModeDeterminate) {
+    if (!animated) {
+      // Update the transition without an animation.
+      [self updateProgressView];
+      return;
+    }
```

Comment:
Consider pulling both the `!animated` logical paths out to an early-exit block at the top of this method. This way you have one less place to have to remember to call the completion block. E.g.:

```objc

    if (!animated) {
      if (_mode == MDCProgressViewModeDeterminate) {
        // Update the transition without an animation.
        [self updateProgressView];
      } else {
        self.progressView.frame = CGRectMake(0, 0, 0, CGRectGetHeight(self.bounds));
        [self updateProgressView];
      }
      if (completion) {
        completion();
      }
      return;
    }